src/Parsers/ParserCreateQuery.cpp

```diff
@@ -353,52 +354,70 @@ bool ParserStorage::parseImpl(Pos & pos, ASTPtr & node, Expected & expected)
     ASTPtr ttl_table;
     ASTPtr settings;
 
-    if (!s_engine.ignore(pos, expected))
-        return false;
+    bool storage_like = false;
```

Comment:
I'm not sure that we should allow to specify other parts of engine definition if engine name is not specified, but maybe it's ok

Reply:
We parse storage definition only in CREATE statements after columns list. I thought it is quite safe to leave this.

Reply:
> only in CREATE statements 

Yes (there are also [1](https://github.com/ClickHouse/ClickHouse/blob/fb9b2d5326933f1887a236455ce574f457ab7f35/src/Interpreters/SystemLog.cpp#L478) and [2](https://github.com/ClickHouse/ClickHouse/blob/b2cfa70541c97f32ee8e0ff72588f418d7fe8e8d/programs/copier/TaskTableAndShard.h#L293), but these places does not look dangerous at the first glance)

> after columns list

Not precisely, we also parse storage definition in `CREATE ... AS another_table` and `CREATE ... AS SELECT` queries that may not have columns list.